src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -8,55 +8,72 @@
 
 namespace Microsoft.Docs.Build
 {
-    internal static class JsonSchemaTransform
+    internal class JsonSchemaTransformer
     {
-        public static (List<Error> errors, JToken token) TransformContent(Document file, Context context, JsonSchema schema, JToken token, Action<Document> buildChild)
+        private readonly JsonSchema _schema;
+        private readonly JsonSchemaDefinition _definitions;
+
+        public JsonSchemaTransformer(JsonSchema schema)
         {
-            var errors = new List<Error>();
+            _schema = schema;
+            _definitions = new JsonSchemaDefinition(schema);
```

Comment:
We are creating a new instance of `JsonSchemaDefinition` each for `JsonSchemaTransformer` and `JsonSchemaValidator`, Is it possible to share 1 instance for both?

Reply:
just to make the interface simple, one extra instance is likely not going to cause perf problems, an instance per file is very likely.